scripts/generate_rust_analyzer.sh

```diff
@@ -0,0 +1,97 @@
+#!/usr/bin/env bash
+
+srctree=$1
+objtree=$2
+lib_src=$3
+bindgen_file=$4
+
+
+# TODO: create issue at RA to accept @arg file
+function generate_cfgs() {
+    echo '['
+    cfgs=$(cat $objtree/include/generated/rustc_cfg | sed 's/--cfg=//' | sed 's/"/\\"/g' | sed 's/$/",/' | sed 's/^/"/')
+    echo ${cfgs::-1}
+    echo ']'
+}
+
+function generate_crate() {
+    name=$1
+    module=$2
+    member=${3:-"true"}
+    cfg=${4:-$(generate_cfgs)}
+    deps=${5:-'[{"crate":0,"name":"core"},{"crate":2,"name":"alloc"},{"crate":4,"name":"kernel"}]'}
+    extra=${CRATE_EXTRA:-""}
+
+    echo "{
+        \"display_name\":\"$name\",
+        \"root_module\":\"$module\",
+        \"edition\":\"2018\",
+        \"deps\":$deps,
+        \"is_workspace_member\": $member,
+        \"cfg\": $cfg,
+        \"env\": {
+            \"RUST_BINDGEN_FILE\": \"$bindgen_file\",
```

Comment:
```suggestion
            \"RUST_BINDINGS_FILE\": \"$bindgen_file\",
```
After correcting the env var and adding something like
```json
        "source": {
                "include_dirs": ["/home/bobo1239/Development/Rust/linux/rust"],
                "exclude_dirs": []
        },
```
to the `kernel` crate in `rust-project.json`, autocomplete works with the generated bindings. (it is somewhat slow though...)

Reply:
To clarify, the `include_dirs` is needed for rust analyzer to include the generated bindings file to its virtual file system. By default it only considers `${srcdir}/rust/kernel` since that's the directory where the `kernel` crate root lies.

I've tried an out-of-tree build now and have noticed that my last `"source"` suggestion wasn't quite correct. It should instead be:
```json
        "source": {
                "include_dirs": ["${srctree}/rust/kernel", "${objtree}/rust"],
                "exclude_dirs": []
        },
```
This will include the sources for `kernel` and also the generated bindings file in the `objtree`.

Reply:
Will try to commit that today. Is that only for the kernel crate?

Reply:
Yes, since that is the crate which includes the `RUST_BINDINGS_FILE`. Also we are actually forced to do this only for the `kernel` crate since the following is stated in the [RA documentation](https://rust-analyzer.github.io/manual.html#non-cargo-based-projects):
```
    /// Different crates can share the same `source`.
    ///
    /// If two crates share an `.rs` file in common,
    /// they *must* have the same `source`.
    /// rust-analyzer assumes that files from one
    /// source can't refer to files in another source.
```